Jenkinsfile

```diff
@@ -1,169 +1,158 @@
-                    branches: [[name: '*/master']],
-                    doGenerateSubmoduleConfigurations: false,
-                    extensions: [[$class: 'CleanCheckout']],
-                    gitTool: 'native git',
-                    submoduleCfg: [],
-                    userRemoteConfigs: [[url: 'https://github.com/realm/realm-java.git']]
-                ])
-            }
-
-            stage 'Docker build'
-            def buildEnv = docker.build 'realm-java:snapshot'
-            buildEnv.inside("--privileged -v /dev/bus/usb:/dev/bus/usb -v ${env.HOME}/gradle-cache:/root/.gradle -v /root/adbkeys:/root/.android") {
-                stage 'JVM tests'
-                try {
-                    gradle 'assemble check javadoc'
-                } finally {
-                    storeJunitResults 'realm/realm-annotations-processor/build/test-results/TEST-*.xml'
-                }
-
-                try {
-                    sh 'cd examples && ./gradlew unitTestExample:check --stacktrace'
-                } finally {
-                    storeJunitResults 'examples/unitTestExample/build/test-results/**/TEST-*.xml'
-                }
-
-                stage 'Static code analysis'
-                try {
-                    sh 'cd realm && ./gradlew findbugs pmd checkstyle --stacktrace'
-                } finally {
-                    publishHTML(target: [allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: 'realm/realm-library/build/findbugs', reportFiles: 'findbugs-output.html', reportName: 'Findbugs issues'])
-                    publishHTML(target: [allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: 'realm/realm-library/build/reports/pmd', reportFiles: 'pmd.html', reportName: 'PMD Issues'])
-                }
-
-                stage 'Run instrumented tests'
-                boolean archiveLog = true
-                String backgroundPid
-                try {
-                    backgroundPid = startLogCatCollector()
-                    sh 'cd realm && ./gradlew connectedUnitTests --stacktrace'
-                    archiveLog = false;
-                } finally {
-                    stopLogCatCollector(backgroundPid, archiveLog)
-                    storeJunitResults 'realm/realm-library/build/outputs/androidTest-results/connected/TEST-*.xml'
-                }
-
-               // TODO: add support for running monkey on the example apps
-
-                if (env.BRANCH_NAME == 'master') {
-                    stage 'Collect metrics'
-                    collectAarMetrics()
-
-                    stage 'Publish to OJO'
-                    sh 'chmod +x gradlew && ./gradlew assemble ojoUpload'
-                }
-            }
+import groovy.json.JsonOutput
+
+def buildSuccess = false
+try {
+  node('docker') {
+    // Allocate a custom workspace to avoid having % in the path (it breaks ld)
+    ws('/tmp/realm-java') {
+      stage 'SCM'
+      checkout scm
```

Comment:
Yes, and in this case it's completely handled by the Github Organisation Folder plugin